### ISSUE TITLE
Psi4 crashes if it's been compiled with -D_GLIBCXX_ASSERTIONS 

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#1255]
The  -D_GLIBCXX_ASSERTIONS has been added to the standard Fedora build flags starting from Fedora 28. This turns on cheap range checks for C++ arrays, vectors, and strings, see
https://fedoraproject.org/wiki/Changes/HardeningFlags28

After a lot of work, I was able to get the Fedora package updated to psi4 1.2.1 and package its new dependencies. However, the new, stringent build flags reveal some broken code in psi4.

Could you build the code with -D_GLIBCXX_ASSERTIONS and fix all the crashing code?

**Comment by susilehtola** [comment]
Indeed, the first crash I saw appears to have been fixed in ca64524802: the c++ vectors weren't initialized properly, using reserve() instead of resize(). Let me see if there are more.

**Comment by raimis** [comment]
`cubeprop` will be likely fixed by #1237

**Comment by susilehtola** [comment]
psimrcc-fd-freq2 crashes at https://github.com/psi4/psi4/blob/81b7428fcbbb96ff583553fc07f388420e3dbe20/psi4/src/psi4/libmoinfo/moinfo.cc#L263

fci-tdm and fci-tdm-2 crash at https://github.com/psi4/psi4/blob/81b7428fcbbb96ff583553fc07f388420e3dbe20/psi4/src/psi4/detci/sem.cc#L994

cubeprop crashes at https://github.com/psi4/psi4/blob/81b7428fcbbb96ff583553fc07f388420e3dbe20/psi4/src/psi4/libfock/PKmanagers.cc#L599

cc46 and cc47 crash at https://github.com/psi4/psi4/blob/81b7428fcbbb96ff583553fc07f388420e3dbe20/psi4/src/psi4/cceom/get_moinfo.cc#L97

**Comment by susilehtola** [comment]
> `cubeprop` will be likely fixed by #1237

Looks like it. Just the three others to go ;)

**Comment by susilehtola** [comment]
Just to be clear - I'm not working on this :clown_face:

**Comment by dgasmith** [comment]
@susilehtola Is this still an issue after the above PR's?

**Comment by susilehtola** [comment]
@dgasmith no idea; the problem is that since Psi4 has been picking up new dependencies left and right I haven't had the opportunity to update the Fedora package...
